e2e-tests/demand-backup-physical-sharded/run

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+
+set -o errexit
+
+test_dir=$(realpath "$(dirname "$0")")
+. "${test_dir}/../functions"
+
+run_backup() {
+	local storage=$1
+	local backup_name=$2
+
+	cat $test_dir/conf/backup.yml \
+		| $sed -e "s/name:/name: ${backup_name}/" \
+		| $sed -e "s/storageName:/storageName: ${storage}/" \
+		| kubectl_bin apply -f -
+}
+
+run_restore() {
+	local backup_name=$1
+
+	cat $test_dir/conf/restore.yml \
+		| $sed -e "s/name:/name: restore-${backup_name}/" \
+		| $sed -e "s/backupName:/backupName: ${backup_name}/" \
+		| kubectl_bin apply -f -
+}
+
+run_recovery_check() {
+	local backup_name=$1
+	local compare_suffix=${2:-"_restore"}
+	local isSharded=${3:-0}
+
+	wait_restore "${backup}" "${cluster}" "requested"
+	echo
+
+	compare_kubectl "statefulset/${cluster}-rs0" ${compare_suffix}
+
+	wait_restore "${backup}" "${cluster}" "ready"
+	kubectl_bin get psmdb ${cluster} -o yaml
+	if [ $(kubectl_bin get psmdb ${cluster} -o yaml | yq '.metadata.annotations."percona.com/resync-pbm"') == null ]; then
+		echo "psmdb/${cluster} should be annotated with percona.com/resync-pbm after a physical restore"
+		exit 1
+	fi
+	echo
+	set -o xtrace
+}
+
+create_infra "${namespace}"
+
+deploy_minio
+apply_s3_storage_secrets
+
+### Case 1: Backup and restore on sharded cluster
+desc 'Testing on sharded cluster'
+
+echo "Creating PSMDB cluster"
+cluster="some-name"
+kubectl_bin apply -f "${conf_dir}/secrets.yml"
+apply_cluster "${test_dir}/conf/${cluster}-sharded.yml"
+kubectl_bin apply -f "${conf_dir}/client_with_tls.yml"
+
+echo "check if all pods started"
+wait_for_running ${cluster}-rs0 3
+wait_for_running ${cluster}-cfg 3
+wait_for_running ${cluster}-mongos 3
+wait_cluster_consistency ${cluster}
+
+run_mongos \
+	'db.createUser({user:"myApp",pwd:"myPass",roles:[{db:"myApp",role:"readWrite"}]})' \
+	"userAdmin:userAdmin123456@${cluster}-mongos.${namespace}"
+sleep 1
+run_mongos \
+	'use myApp\n db.test.insert({ x: 100501 })' \
+	"myApp:myPass@${cluster}-mongos.${namespace}"
+sleep 5
+compare_mongos_cmd "find" "myApp:myPass@${cluster}-mongos.${namespace}" "-sharded"
+
+echo 'running backups'
+backup_name_minio="backup-minio-sharded"
+if [ -z "$SKIP_BACKUPS_TO_AWS_GCP_AZURE" ]; then
+	backup_name_aws="backup-aws-s3-sharded"
+	backup_name_gcp="backup-gcp-cs-sharded"
+	backup_name_azure="backup-azure-blob-sharded"
+fi
+
+if [ -z "$SKIP_BACKUPS_TO_AWS_GCP_AZURE" ]; then
+	run_backup aws-s3 ${backup_name_aws}
+	run_backup gcp-cs ${backup_name_gcp}
+	run_backup azure-blob ${backup_name_azure}
+fi
+run_backup minio ${backup_name_minio}
+
+if [ -z "$SKIP_BACKUPS_TO_AWS_GCP_AZURE" ]; then
+	wait_backup "${backup_name_aws}"
+	wait_backup "${backup_name_gcp}"
+	wait_backup "${backup_name_azure}"
+fi
+wait_backup "${backup_name_minio}"
```

Comment:
could we put all connected with minio one by one and then just add one if for external backups 
```
if [ -z "$SKIP_BACKUPS_TO_AWS_GCP_AZURE" ]; then
	backup_name_aws="backup-aws-s3-sharded"
	backup_name_gcp="backup-gcp-cs-sharded"
	backup_name_azure="backup-azure-blob-sharded"

	run_backup aws-s3 ${backup_name_aws}
	run_backup gcp-cs ${backup_name_gcp}
	run_backup azure-blob ${backup_name_azure}
	
        wait_backup "${backup_name_aws}"
	wait_backup "${backup_name_gcp}"
	wait_backup "${backup_name_azure}"
fi
```

Reply:
Thanks, applied.